homeassistant/components/todoist/calendar.py

```diff
@@ -562,14 +560,16 @@ def select_best_task(project_tasks):
                 continue
 
             if proposed_event[PRIORITY] == event[PRIORITY] and (
-                proposed_event[END] < event[END]
+                event[END] is not None and proposed_event[END] < event[END]
```

Comment:
Added an additional check as mypy pointed out this value could be `None`.

Reply:
Interesting because there are other checks above like `event[DEND].date()`...

Reply:
Consider if this gets simpler with some local variables for proposed_event_end and event_end

Reply:
I don't think mypy is clever enough to figure out that if `event[END]` is `None` that we will never reach this block: https://github.com/home-assistant/core/pull/84210/files/14e27151ca85b347a610279310035ae35cc793c4#diff-0df90a93485af80ce0879138de9155f75b1a60a63454406902e6246df2da2194R543-R546.

mypy is instead inferring the type from the `TodoistEvent` typed dict which says the `end` key could be `datetime | None`.

Reply:
I think the loop is confusing it, otherwise I think it usually can tell when you have `is None` guards. Anyway, a local variable might make it easier, but also adds more cruft.